labgrid/remote/client.py

```diff
@@ -1531,6 +1572,24 @@ def main():
     tmc_subparser.add_argument('action', choices=['info', 'values'])
     tmc_subparser.set_defaults(func=ClientSession.tmc_channel)
 
+    subparser = subparsers.add_parser('xlx', help="connect to a Xilinx Vivado hardware server")
```

Comment:
In the long run we'll want to support tools like [openfpgaloader](https://github.com/trabucayre/openFPGALoader), this should rather be called `fpga`.

Reply:
Making the interface vendor-neutral sounds like a good idea; the `xlx` subparser was more like a first try.

I have the following two high-level `fpga` subcommands in mind:
- `program-bitstream`: Program bitstream via `xsdb` (Xilinx), `quartus_pgm` (Intel) or `openFPGALoader` (vendor-neutral) into volatile memory
- `program-flash`: Program data (bitstream and software for SoCs) via `program_flash` (Xilinx), `quartus_hps` (Intel) or `openFPGALoader` (vendor-neutral) into non-volatile memory

In additon, there could be vendor-specific, lower level commands like `xsdb` for the Xilinx System Debugger Tcl-based shell.

I only have first-hand experience with the Xilinx toolchain; so feel free to fill in the gaps here (especially on the Lattice side).

Reply:
I have implemented the interface as proposed.